### PR TITLE
Comply with flake8 linter

### PR DESCRIPTION
**Pull request by JoshuaWatt**
Reformats Python code according to pep8, and fixes up flake8 warnings.

Adds a lint script that can be run to fix up the code to be conformant
in the future.

Updates the Travis CI tests to run flake8 as part of the automated tests

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,2 +1,2 @@
-[pycodestyle]
```
Comment (jamesharris-garmin) [inline setup.cfg]: Do we want to also support pydocstyle?

Comment (JoshuaWatt) [reply setup.cfg]: Possibly. I don't think we have any docstrings right now anyway, and I think that could be a separate patch.

**Comment by jamesharris-garmin** [comment]
Why did our code coverage go down?

**Comment by JoshuaWatt** [comment]
> Why did our code coverage go down?

I think because I removed 3 unused imports which were previously covered. This lowers the overall ratio of covered to uncovered lines... I'm not really worried about it

**Comment by jamesharris-garmin** [comment]
I am alright with this patch then. We should put up another patch to use the tool tox to handle test execution: https://tox.readthedocs.io/en/latest/ we could put the downloading of external requirements in there the all the developer would need to do is run "tox" to execute.

**Pull request merged**